Frameworks/CoreFoundationAdditions/_NSCFTemporaryRootObject.mm

```diff
@@ -16,4 +16,8 @@
 #include "_NSCFTemporaryRootObject.h"
 
 @implementation _NSCFTemporaryRootObject
+- (void)doesNotRecognizeSelector:(SEL)selector {
+    // According to the reference platform documentation, this method must never return.
+    abort();
```

Comment:
We could throw a `std::runtime_error` with a known `what`, since we have access to the CRT. It would be just as catchable and generate a trace, I think. What do you think?

That said, if this is being called, something far far more egregious has failed. It'll only happen when Foundation doesn't load and stomp the superclass for the types derived herefrom, or something in CF uses Objective-C in a bad way before that happens.

---
In reply to: [71575314](https://github.com/Microsoft/WinObjC/pull/671#discussion_r71575314) [](ancestors = 71575314,71574896,71574472)